docs/root/operations/performance.rst

```diff
@@ -49,3 +49,24 @@ the event dispatcher for each worker thread has a statistics tree rooted at
   poll_delay_us, Histogram, Polling delays in microseconds
 
 Note that any auxiliary threads are not included here.
+
+.. _operations_performance_watchdog:
+
+Watchdog
+--------
+
+In addition to event loop statistics, Envoy also include a configurable
+:ref:`watchdog <envoy_api_field_config.bootstrap.v2.Bootstrap.watchdog>` system that can increment
+statistics when Envoy is not responsive and optionally kill the server. The statistics are useful
+for understanding at a high level whether Envoy's event loop is not responsive either because it is
+doing too much work, blocking, or not being scheduled by the OS.
+
+The watchdog emits statistics in both the *server.* and *server.<thread_name>.* trees.
```

Comment:
Suggestion (TIOLI) for parity with the event loop stats: the main thread's watchdog stats should look like `server.watchdog.{miss, mega_miss}`, etc. and the worker threads' watchdog stats should look like `listener_manager.worker_<id>.watchdog.{miss, mega_miss}`.

Reply:
I could go either way on this. I was mainly just trying to be consistent with the watchdog stats we have today. @jmarantz any opinion?

Reply:
Oh, it's probably more important to stay consistent with existing watchdog stats.